### ISSUE TITLE
FE cleans up SEC_CLIENT_* config, breaking OSPool auth

### ISSUE THREAD

**Issue by rynge** [glideinWMS/glideinwms#95]
I'm not sure how general this problem is, so I will open an issue instead of a PR. For the OSPool FE, we have mostly switched to IDToken auth. Schedds are issues a token via the OSG token service, which means that the schedd can authenticate to the pool, but the CM does not have a way to authenticate to the schedd. Thus, the FE has to query the schedd anonymously. We have this configured:

```
SEC_CLIENT_AUTHENTICATION = OPTIONAL
SEC_CLIENT_ENCRYPTION = OPTIONAL
SEC_CLIENT_INTEGRITY = OPTIONAL
```

However, the frontend rewrites the config before querying and breaks anonymous querying. To fix this problem, OSPool is currently disabling this via the following patch:

```
diff --git a/creation/lib/cvWCreate.py b/creation/lib/cvWCreate.py
index 00c8efd9..38f2c2a3 100644
--- a/creation/lib/cvWCreate.py
+++ b/creation/lib/cvWCreate.py
@@ -196,7 +196,9 @@ def filter_unwanted_config_attrs(attrs):
         unwanted_attrs.append('TOOL.SEC_%s_INTEGRITY' % context)
 
         # Keep default setting for following
-        if context!="DEFAULT":
+        # Exclude CLIENT as those settings are used to query schedds
+        # and some pools use non-authenticated queries
+        if context not in ["DEFAULT", "CLIENT"]:
             unwanted_attrs.append('SEC_%s_AUTHENTICATION' % context)
             unwanted_attrs.append('SEC_%s_AUTHENTICATION_METHODS' % context)
             unwanted_attrs.append('SEC_%s_INTEGRITY' % context)
```

I think the gwms team should consider if this going to be a general problem when we are moving more and more towards token auth.

**Comment by brianhlin** [comment]
I'm pretty sure @bbockelm has NACK'ed all suggestions to make auth optional. Can we just ask AP admins to get a host cert and depend on SSL auth?

**Comment by bbockelm** [comment]
Yup -- the AP admins either need to allow anonymous secure queries (e.g., SSL auth) or create a token for the frontend to query.

**Comment by rynge** [comment]
@bbockelm I thought this was just a stop gab until HTCondor would do this with some kind of session between the schedd and cm?

From past experience, the SSL idea (or having access point admins provide idtokens to the cm) is not going to work long term - if it is not in the critical path (or properly monitored), it will decay and stop working.
